src/Raven.Server/Utils/Metrics/MeterMetric.cs

```diff
@@ -131,42 +104,167 @@ public void Mark()
+                ["Count"] = Count,
+                ["MeanRate"] = Math.Round(MeanRate, 1),
+                ["OneMinuteRate"] = Math.Round(OneMinuteRate, 1),
+                ["FiveMinuteRate"] = Math.Round(FiveMinuteRate, 1),
+                ["FifteenMinuteRate"] = Math.Round(FifteenMinuteRate, 1)
             };
 
             if (allResults == false)
-                return r;
-
-            var results = new DynamicJsonValue();
-            r["Raw"] = results;
-            var index = Volatile.Read(ref _index) % _m15Rate.Length;
-            var current = DateTime.UtcNow;
-            var now = new TimeSpan(current.Hour, current.Minute, current.Second);
-            var oneSec = TimeSpan.FromSeconds(1);
-            for (int i = index; i >= 0; i--)
+                return result;
+
+            var nowNs = _now();
+            var nowUtc = DateTime.UtcNow;
+            var bucketSeconds = BucketDurationNanoseconds / (double)Clock.NanosecondsInSecond;
+
+            var buckets = new List<(long Start, double Rate)>();
+            for (int i = 0; i < BucketCount; i++)
+            {
+                ref var bucket = ref _buckets[i];
+                var start = Volatile.Read(ref bucket.Start);
+                if (start == 0)
+                    continue;
+
+                if (nowNs - start > MaxWindowNanoseconds)
+                    continue;
+
+                var count = Volatile.Read(ref bucket.Count);
+                if (filterEmpty && count == 0)
+                    continue;
+
+                buckets.Add((start, count / bucketSeconds));
+            }
+
+            buckets.Sort((a, b) => a.Start.CompareTo(b.Start));
+
+            // Emit raw 100ms samples so diagnostics can plot the exact workload shape (no smoothing).
+            var raw = new DynamicJsonValue();
+            foreach (var (start, rate) in buckets)
+            {
+                var ageSeconds = (nowNs - start) / (double)Clock.NanosecondsInSecond;
+                var timestamp = nowUtc - TimeSpan.FromSeconds(ageSeconds);
+                raw[timestamp.ToString("O", CultureInfo.InvariantCulture)] = Math.Round(rate, 1);
+            }
+
+            result["Raw"] = raw;
+            return result;
+        }
+
+        // A small CAS loop guards bucket advancement to avoid the "clear-after-publish" race.
+        // Protocol:
+        // - Writers claim a bucket by CAS-ing Start to a sentinel value (ClaimedStart).
+        // - Under the claim, we zero Count/DurationSum, then publish the real Start via a release write.
+        // - Readers acquire Start and skip buckets with Start <= 0 (never used or claimed/in-progress).
+        // This ensures no thread observes half-initialized state. The loop runs only at quantum boundaries.
+        private void AddToBucket(long timestamp, long value, long duration)
+        {
+            var quantum = timestamp / BucketDurationNanoseconds;
+            var bucketStart = quantum * BucketDurationNanoseconds;
```

Comment:
This is important because makes the `bucketStart` comparable between the timestamps